examples/cqt2midi.py

```diff
@@ -0,0 +1,134 @@
+
+    Returns
+    -------
+    piano_roll : np.ndarray, shape=(128,frames), dtype=int
+        Piano roll representation on audio.
+
+    '''
+    piano_roll = np.abs(cqt)
+    piano_roll = np.digitize(piano_roll, np.linspace(piano_roll.min(),
+                                                     piano_roll.max(), 127))
+    piano_roll[piano_roll < threshold] = 0
+    piano_roll = np.pad(piano_roll,
+                        [(128-max_midi, min_midi), (0, 0)], 'constant')
+    return piano_roll
+
+
+def piano_roll_to_pretty_midi(piano_roll, fs=100, program=0):
+    '''Convert a Piano Roll array into a PrettyMidi object
+     with a single instrument.
+
+    Parameters
+    ----------
+    piano_roll : np.ndarray, shape=(128,frames), dtype=int
+        Piano roll of one instrument
+    fs : int
+        Sampling frequency of the columns, i.e. each column is spaced apart
+        by ``1./fs`` seconds.
+    program : int
+        The program number of the instrument.
+
+    Returns
+    -------
+    midi_object : pretty_midi.PrettyMIDI
+        A pretty_midi.PrettyMIDI class instance describing
+        the piano roll.
+
+    '''
+    notes, frames = piano_roll.shape
+    pm = pretty_midi.PrettyMIDI()
+    instrument = pretty_midi.Instrument(program=program)
+
+    # pad 1 column of zeros so we can acknowledge inital and ending events
+    piano_roll = np.pad(piano_roll, [(0, 0), (1, 1)], 'constant')
+
+    # use changes in velocities to find note on / note off events
+    velocity_changes = np.nonzero(np.diff(piano_roll).T)
+
+    # keep track on velocities and note on times
+    prev_velocities = np.zeros(notes, dtype=int)
+    note_on_time = np.zeros(notes)
+
+    for time, note in zip(*velocity_changes):
+        # use time + 1 because of padding above
+        velocity = piano_roll[note, time + 1]
+        time = time / fs
+        if velocity > 0:
+            if prev_velocities[note] == 0:
+                note_on_time[note] = time
+                prev_velocities[note] = velocity
+        else:
+            pm_note = pretty_midi.Note(
+                    velocity=prev_velocities[note],
+                    pitch=note,
+                    start=note_on_time[note],
+                    end=time)
+            instrument.notes.append(pm_note)
+            prev_velocities[note] = 0
+    pm.instruments.append(instrument)
+    return pm
+
+if __name__ == '__main__':
+    # Set up command-line argument parsing
+    parser = argparse.ArgumentParser(
+        description='Transcribe Audio file to MIDI file',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument('input_audio', action='store',
+                        help='Path to the input Audio file')
+    parser.add_argument('output_midi', action='store',
+                        help='Path where the transcribed MIDI will be written')
+    parser.add_argument('--sr', default=22050, type=int, action='store',
```

Comment:
I think it's OK to just force using librosa's default here for the sampling rate, because if it's changed from 22050 various parameters (hop size etc) of the CQT should also be changed.  Also simplifies things a little.